ts/Stock/StockTools/StockToolsGui.ts

```diff
@@ -145,38 +144,46 @@ function onChartBeforeRedraw(
     this: Chart
 ): void {
     if (this.stockTools) {
-        const optionsChart = this.options.chart as ChartOptions;
-        const listWrapper = this.stockTools.listWrapper,
-            offsetWidth = listWrapper && (
+        this.stockTools.redraw();
+
+        if (this.stockTools.guiEnabled) {
```

Comment:
Q: What do you think about extracting the whole code below to a function, eg `setOffset()` or `calculateOffset()`?